### PR TITLE
feature/audio seek

### PR DESCRIPTION
**Pull request by JarbasAl**
## Description
adds seek functionality to audio service

only implemented in vlc audio backend

as a follow up the playback skill should also be updated

## How to test

use the new methods in audioservice

        from mycroft.skills.audioservice import AudioService

        audio = AudioService(bus)
        audio.play(track)
        audio.seek(10)  # skip 10 seconds
        audio.seek(-10)  # rewind 10 seconds
        audio.seek_forward(20)  # skip 20 seconds
        audio.seek_backward(5)  # rewind 5 seconds

## Contributor license agreement signed?
CLA [affirmative] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/skills/audioservice.py [hunk 1]
```diff
@@ -121,6 +121,23 @@ def resume(self):
         """ Resume paused playback. """
         self.bus.emit(Message('mycroft.audio.service.resume'))
 
+    def seek(self, seconds=1):
+        """ seek X seconds """
```
Comment (forslund) [inline mycroft/skills/audioservice.py]: Add the *Arguments:* section to these docstrings

File: mycroft/audio/audioservice.py [hunk 2]
```diff
@@ -187,6 +187,8 @@ def load_services_callback(self):
         self.bus.on('mycroft.audio.service.next', self._next)
         self.bus.on('mycroft.audio.service.prev', self._prev)
         self.bus.on('mycroft.audio.service.track_info', self._track_info)
+        self.bus.on('mycroft.audio.service.seek.forward', self._seek_forward)
+        self.bus.on('mycroft.audio.service.seek.backward', self._seek_backward)
```
Comment (forslund) [inline mycroft/audio/audioservice.py]: I think `mycroft.audio.service.seek_forward` and `mycroft.audio.service.seek_backward` will fit better with the others.

File: mycroft/audio/services/__init__.py [hunk 3]
```diff
@@ -15,7 +15,7 @@
 from abc import ABCMeta, abstractmethod
 
 
-class AudioBackend():
+class AudioBackend(object):
```
Comment (forslund) [inline mycroft/audio/services/__init__.py]: No need to inherit from object in python3

Comment (JarbasAl) [reply mycroft/audio/services/__init__.py]: isn't that the standard however? we are doing it everywhere else

File: mycroft/audio/audioservice.py (hunk 2 above)
Comment (JarbasAl) [reply mycroft/audio/audioservice.py]: agreed, will make changes

File: mycroft/skills/audioservice.py (hunk 1 above)
Comment (JarbasAl) [reply mycroft/skills/audioservice.py]: :+1: forgot about that

File: mycroft/audio/services/__init__.py (hunk 3 above)
Comment (forslund) [reply mycroft/audio/services/__init__.py]: Yeah I've noticed it was still hanging around. Added a PR to remove them.

Comment (JarbasAl) [reply mycroft/audio/services/__init__.py]: see https://github.com/MycroftAI/mycroft-core/pull/1945#issuecomment-453237439

either way i do not oppose this change, when #1945 is merged i'll update this

**Comment by JarbasAl** [comment]
should probably add a check in vlc to ensure time is not negative

and also that it isn't higher than audio length, need to dig into vlc api for this one

**Comment by forslund** [comment]
Tested and works really well. I went ahead and rebased it. Merging now.

Many thanks @JarbasAI !

**Pull request merged**